Shared/Samples/Show viewshed from camera in scene/ShowViewshedFromCameraInSceneView.swift

```diff
@@ -0,0 +1,130 @@
+            }
+        }
+    }
+    
+    private var viewshedButton: some View {
+        Button("Viewshed from here") {
+            // Update viewshed based on current camera location when button is tapped.
+            model.updateViewshedFromCurrentCamera()
+        }
+        .buttonStyle(.borderedProminent)
+        .tint(.purple)
+        .padding(.horizontal)
+    }
+}
+
+private extension ShowViewshedFromCameraInSceneView {
+    @MainActor
+    @Observable
+    final class Model {
+        var currentCamera: Camera? = .initialCamera
+        /// An analysis overlay used to display the viewshed analysis visualization.
+        let analysisOverlay = AnalysisOverlay()
+        /// The viewshed which is updated by the camera.
+        @ObservationIgnored private var viewshed: LocationViewshed?
+        /// 3D Scene setup with imagery basemap, elevation, and mesh layer.
+        let scene: ArcGIS.Scene = {
+            let scene = Scene(basemapStyle: .arcGISImagery)
+            scene.baseSurface.addElevationSource(
+                ArcGISTiledElevationSource(url: .elevation)
+            )
+            scene.addOperationalLayer(
+                IntegratedMeshLayer(url: .gironaMeshService)
+            )
+            return scene
+        }()
+        
+        /// Applies a viewshed at the initial camera position.
+        func setInitialViewshed() {
+            guard let camera = currentCamera else { return }
+            setupViewshed(using: camera)
+        }
+        
+        /// Updates the viewshed using the current camera position.
+        func updateViewshedFromCurrentCamera() {
+            guard let camera = currentCamera else { return }
+            viewshed?.update(from: camera)
+        }
+        
+        /// Applies the viewshed to the scene and sets the UI for analysis.
+        private func setupViewshed(using camera: Camera) {
+            // Create and configure the new viewshed.
+            let newViewshed = LocationViewshed(
+                camera: camera,
+                minDistance: 1.0,
+                maxDistance: 1000.0
+            )
+            // Set visual appearance of the viewshed.
+            Viewshed.visibleColor = UIColor.green.withAlphaComponent(0.5)
+            Viewshed.obstructedColor = UIColor.red.withAlphaComponent(0.5)
+            // Add the new viewshed to the overlay.
+            analysisOverlay.addAnalysis(newViewshed)
+            viewshed = newViewshed
+        }
+    }
+}
+
+private extension Camera {
+    
+    /// Camera that is 200m above ground over Girona, Spain,
+    /// looking slightly downward at the landscape.
+    static var initialCamera: Camera {
+        Camera(
+            location: Point(
+                x: 2.8214,
+                y: 41.985,
+                z: 200.0,
+                spatialReference: .wgs84
+            ),
+            heading: 332.131,
+            pitch: 82.4732,
+            roll: 0
+        )
+    }
+}
+
+private extension URL {
```

Comment:
I don't see the value of these static vars also because they are only being used in one spot. So we could just declare them at the point of use. Maybe this is a common pattern in the Samples? @yo1995

Reply:
```
        let scene: ArcGIS.Scene = {
            let scene = Scene(basemapStyle: .arcGISImagery)
            scene.baseSurface.addElevationSource(
                ArcGISTiledElevationSource(
                    url: URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
                )
            )
            scene.addOperationalLayer(
                IntegratedMeshLayer(
                    url: URL(string: "https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/Girona_3D/SceneServer")!
                )
            )
            return scene
        }()
```

@zkline101 @yo1995 would this work better?

Reply:
I've no preference. Either is fine.